timesketch/lib/analyzers/win_crash.py

```diff
@@ -0,0 +1,195 @@
+            'data_type:"fs:stat"',
+            'filename:"/Microsoft/Windows/WER/"',
+            'filename:/((Non)?Critical|AppCrash)_.*/',
+            'file_entry_type:("directory" or "2")',
+        ),
+        'Registry - Crash Reporting' : (
+            'data_type:"windows:registry:key_value"',
+            r'key_path:"\\Control\\CrashControl"',
+            'values:("LogEvent: REG_DWORD_LE 0"' + \
+            ' OR "SendAlert: REG_DWORD_LE 0"' + \
+            ' OR "CrashDumpEnabled: REG_DWORD_LE 0")',
+        ),
+        'Registry - Error Reporting' : (
+            'data_type:"windows:registry:key_value"',
+            r'key_path:"\\Software\\Microsoft\\PCHealth\\ErrorReporting"',
+            'values:("DoReport: REG_DWORD_LE 0" OR "ShowUI: REG_DWORD_LE 0")',
+        ),
+    }
+
+    def __init__(self, index_name, sketch_id):
+        """Initialize The Sketch Analyzer.
+
+        Args:
+            index_name: Elasticsearch index name
+            sketch_id: Sketch ID
+        """
+        self.index_name = index_name
+        super(WinCrashSketchPlugin, self).__init__(index_name, sketch_id)
+
+    def formulate_query(self, elements):
+        """Generates the Elasticsearch query.
+
+        Args:
+            elements: Dictionary with a list of conditions
+
+        Returns:
+            The Elasticsearch query
+        """
+        conditions = list()
+        for e in elements.values():
+            conditions += ['({0})'.format(' AND '.join(e))]
+        return ' OR '.join(conditions)
+
+    def extract_filename(self, text):
+        """Finds filenames of crashed applications using a regular expression.
+
+        Args:
+            text: String that might contain the filename
+
+        Returns:
+            The string with filename if found
+        """
+        if '.exe' in str(text or '').lower():
+            match = self.FILENAME_REGEX.search(text)
+            if match:
+                # The regex can match on full file paths and filenames,
+                # so only return the filename.
+                return min([m for m in match.groups() if m])
+        return None
+
+    def mark_as_crash(self, event, filename):
+        """Mark entries with crash artefacts.
+
+        Args:
+            event: Elasticsearch event
+            filename: Application that crashed
+        """
+        if filename:
+            event.add_attributes({'crash_app': filename})
+        event.add_tags(['win_crash'])
+
+    def run(self):
+        """Entry point for the analyzer.
+
+        Returns:
+            String with summary of the analyzer result
+        """
+        query = self.formulate_query(self.QUERY_ELEMENTS)
+
+        return_fields = ['data_type', 'message', 'filename']
+
+        # Generator of events based on your query.
+        events = self.event_stream(
+            query_string=query, return_fields=return_fields)
+
+        # Container for filenames of crashed applications.
+        filenames = list()
```

Comment:
why have this as a list only to do `set` operation on it at the end? why ont have this as a set to begin with

```
...
filenames = set()
...
filenames.add(filename)
```

or have this as a `collections.Counter` if you want to have both a unique set of filenames and a count of them (but only if you want to use the count\)

Reply:
Good point, I replaced list() with set().